### DISCUSSION TITLE
Expected time/epoch for conceptual captions (R50)

### DISCUSSION THREAD

**Discussion by piotr-teterwak** [mlfoundations/open_clip#55] [Q&A]
How long is a reasonable time for an epoch using 8 workers?   I'm seeing about 8 hours/epoch, for the resnet50.  Launch command from the README:

``` python -u src/training/main.py \
    --save-frequency 1 \
    --zeroshot-frequency 1 \
    --report-to tensorboard \
    --train-data="/path/to/train_data.csv"  \
    --val-data="/path/to/validation_data.csv"  \
    --csv-img-key filepath \
    --csv-caption-key title \
    --imagenet-val=/path/to/imagenet/root/val/ \
    --warmup 10000 \
    --batch-size=128 \
    --lr=1e-3 \
    --wd=0.1 \
    --epochs=30 \
    --workers=8 \
    --model RN50
```

Thank you!

**Comment by carlini** [comment]
When I run on a machine with 8xV100 using the settings from the README, it takes about 12 hours to finish training 30 epochs.

**Comment by rwightman** [comment]
Also, 4xV100 (32 GB) was 3 days for cc12m

Moving this to discussion for future reference

**Reply by yao-eastside** [reply]
Very helpful. I have twice the training size (20m), and twice the number of GPUs (8 v100 GPUs), it takes 2 hours / epoch. So total is 60 hours, which is 2.5 days. It is quite consistent with your time.

**Comment by yao-eastside** [comment]
FYI, for 8 v100 GPUs, (4 nodes), it takes 2 hours per epoch for training size of 20m using RN50.

@piotr-teterwak, you did not mention your training size.

I think number of GPUs is more important than number of workers. If your GPU is 100% already, further increasing number of workers won't help. To speed up, you will want to increase the number of GPUs.
